resonate/scheduler.py

```diff
@@ -507,7 +522,12 @@ def apply(self, cmd: Command) -> None:
                 assert (promise is not None) == opts.durable, "Promise must be set iff durable."
 
                 cls = Coro if isgeneratorfunction(func) else Lfnc
-                self.graph.root.transition(Enabled(Running(cls(id, self.id, conv, timeout, func, args, kwargs, opts, self.ctx).map(promise=promise))))
+                func = cls(id, self.id, conv, timeout, func, args, kwargs, opts, self.ctx).map(promise=promise)
+
+                if promise and promise.completed:
+                    self.graph.root.transition(Enabled(Completed(func)))
```

Comment:
This was a cool find by DST. It is possible to receive an invoke with a completed promise. It seems to happen most often when the root promise has timed out.